runtime/src/accounts_db.rs

```diff
@@ -4963,13 +4964,25 @@ impl AccountsDb {
         }
     }
 
-    pub fn get_snapshot_storages(&self, snapshot_slot: Slot) -> SnapshotStorages {
+    pub fn get_snapshot_storages(
+        &self,
+        snapshot_slot: Slot,
+        ancestors: Option<&Ancestors>,
```

Comment:
@sakridge, it looks like we have to know ancestors here for getting storages, at least for tests. Stores are in slots that aren't in the roots set. I haven't thought this all the way through yet, but I figured I'd get feedback since it just passed all tests for the first time. This is related to the startup time issue of checking cap.

Reply:
Yes. This seems right. It does the capitalization check on the newest bank after replaying all of what is in blockstore which may or may not be rooted.